src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+        nonwear_epoch1, physical_activity_levels and sleep_windows_epoch1. nonwear_array
+        values are in 15 minute blocks and are upsampled to match epoch1 time for the
+        purposes of saving. Additionally the sleep window data is a list of timestamp
+        pairs, which are used to create a binary array in epoch1 time. Two files will
+        be saved, one of each temporal resolution. This means if output is entered as
+        /path/to/file/file.csv,The file names will be labeled as file_epoch1.csv and
+        file_raw_time.csv in the path/to/file directory.
+
+        Args:
+            output: The path and file name of the data to be saved. as either a csv or
+                parquet files.
+
+        Returns:
+            None.
+
+        Raises:
+            InvalidFileTypeError: If the output file path ends with any extension other
+                than csv or parquet.
+        """
+        if output.suffix not in [".csv", ".parquet"]:
+            raise InvalidFileTypeError(
+                f"The extension: {output.suffix} is not supported.",
+                "Please save the file as .csv or .parquet",
+            )
+
+        results_epoch1_time = self._results_to_dataframe()
+        results_raw_time = self._results_to_dataframe(use_epoch1_time=False)
+
+        save_path_epoch1 = output.with_name(output.stem + "_epoch1" + output.suffix)
+        save_path_raw_time = output.with_name(output.stem + "_raw_time" + output.suffix)
+
+        if output.suffix == ".csv":
+            results_epoch1_time.write_csv(save_path_epoch1, separator=",")
+            results_raw_time.write_csv(save_path_raw_time, separator=",")
+        elif output.suffix == ".parquet":
+            results_epoch1_time.write_parquet(save_path_epoch1)
+            results_raw_time.write_parquet(save_path_raw_time)
+
+    def _results_to_dataframe(self, use_epoch1_time: bool = True) -> pl.DataFrame:
+        """Format results into a dataframe.
+
+        Args:
+            use_epoch1_time: Determines which temporal resolution and data to use. If
+                True the dataframe will contain enmo_epoch1, anglez_epoch1,
+                nonwear_epoch1, physical_activity_levels and sleep_windows_epoch1. If
+                False the the dataframe will be in "raw" time and contain the enmo and
+                anglez data.
+
+        Returns:
+            The polars dataframe to be saved into csv or parquet format.
+
+        Raises:
+            ValueError if the attribute for the time column is None.
+        """
+        data_source = self.enmo_epoch1 if use_epoch1_time else self.enmo
+
+        if data_source is None:
+            raise ValueError(f"{data_source} is None, can't construct time column.")
+
+        time_data = data_source.time
+
+        results_dataframe = pl.DataFrame({"time": time_data})
+
+        for field_name, field_value in self:
+            if field_value is not None:
```

Comment:
Why would a field be none?

Reply:
I was getting mypy errors here as I have the fields as Optional, and defaulting to None, so explicitly ruling that out here was how I addressed it.